tools/tfuser/cmds_network.go

```diff
@@ -214,23 +394,311 @@ func isIn(l []uint, i uint) bool {
 	return false
 }
 
-func generatePeers(nodeID string, n pkg.Network) []pkg.Peer {
-	peers := make([]pkg.Peer, 0, len(n.NetResources))
+func hasIPv4(n pkg.NetResource) bool {
+	for _, pep := range n.PubEndpoints {
+		if pep.To4() != nil {
+			return true
+		}
+	}
+	return false
+}
+
+// This function assumes:
+// - that a hidden node has functioning IPv4
```

Comment:
If the node has IPv6, it should have a public IPv6 endpoint. Since the node has no public endpoints ("hidden") , it can only have a (natted) IPv4. Or it has no internet access, but then it couldn't have registered itself in the first place. Also not sure what you mean by "kind of ipv4"